Added support for qemu-zynq7 (Armv7)  VM

Raagul1 opened the pull request:
For FRC Project, Currently creating image from Roborio is a manual process, Which requires Hardware and also a time consuming process. I'm trying to automate this image creation process with qemu to eliminate the necessity of Roborio hardware and also to make the process faster.
Support has been added for Qemu VM, None of existing machine configuration has been modified, I have tested building it in my local machine with these file changes
Find the below link to track this NI-internal work items
[Feature 1961177](https://dev.azure.com/ni/DevCentral/_workitems/edit/1961177): Automate the image creation process for FRC

amstewart commented on conf/bblayers.conf.sample:
```diff
@@ -31,6 +31,8 @@ BBLAYERS ?= " \
 	${GIT_REPODIR}/meta-security \
 	${GIT_REPODIR}/meta-selinux \
 	${GIT_REPODIR}/meta-virtualization \
+	${GIT_REPODIR}/meta-xilinx/meta-xilinx-bsp \
+	${GIT_REPODIR}/meta-xilinx/meta-xilinx-contrib \
```
Before we can add this meta-layer to the bblayers, we'll first need to:
1. fork the upstream meta-xilinx repo into the [NI GH org](https://github.com/ni/).
2. add the layer repo to the [nilrt.git .gitmodules](https://github.com/ni/nilrt/blob/nilrt-academic/master/sumo/.gitmodules) file.

I'll start an internal discussion with the RTOS team about if we are willing to accept the maintenance burden of that repo.

amstewart replied:
I've requested that the meta-xilinx repo be added to the NI org.

amstewart commented on conf/local.conf.sample:
```diff
@@ -16,7 +16,7 @@
 #
 # Machine Selection
-#   Supported NILRT machine values are: xilinx-zynq
+#   Supported NILRT machine values are: xilinx-zynq, qemu-zynq7
```
I think this change will cause BB to start using [the qemu-arm7.conf](https://git.yoctoproject.org/meta-xilinx/tree/meta-xilinx-bsp/conf/machine/qemu-zynq7.conf?h=sumo) provided by meta-xilinx. I think you'll need to reconcile this with the NILRT-specific settings in the meta-nilrt [xilinx-zynqhf.conf](https://github.com/ni/meta-nilrt/blob/nilrt-academic/master/sumo/conf/machine/xilinx-zynqhf.conf) machine configs. You might need to author your own wrapper config.

Raagul1 replied:
I removed this change from conf.sample file.

amstewart replied:
The changed comment doesn't really matter. I mean that, when you specify `DISTRO=qemu-zynq7`, it will change the MACHINE .conf file that bitbake uses from [`xilinx-zynqhf.conf`](https://github.com/ni/meta-nilrt/blob/nilrt-academic/master/sumo/conf/machine/xilinx-zynqhf.conf) (which we have modified in the meta-nilrt layer, to the `qemu-arm7.conf` we get from upstream OE-core.

You should be sure that the values that we define in the zynqhf.conf aren't needed in your QEMU build.

amstewart replied:
I'm going to drop this issue, since it isn't likely to break anything about the academic build, even if it is wrong.

amstewart reviewed (changes requested):
Hey @Raagul1, a few things:

We generally make changes at the commit-level in OE (rather than the PR-level). So could you...
1. squash your commits in this PR - since they are only really accomplishing a single change.
2. add some detail to your commits describing: the current state of the codebase, what you are changing, and why you are changing it.

Could you also
1. sign your commits using `git commit -s`.
2. add some detail to your PR description about:
    1. What use case you're trying to address and why this PR addresses it. I assume this is for FRC, but is it for testing or to serve some other requirement?
    2. What testing have you done to verify that these changes are buildable and do not break the academic mainline?
3. link to any NI-internal work items that track this work.

ghost commented:
> Hey @Raagul1, a few things:
> 
> We generally make changes at the commit-level in OE (rather than the PR-level). So could you...
> 
> 1. squash your commits in this PR - since they are only really accomplishing a single change.
> 2. add some detail to your commits describing: the current state of the codebase, what you are changing, and why you are changing it.
> 
> Could you also
> 
> 1. sign your commits using `git commit -s`.
> 2. add some detail to your PR description about:
>    
>    1. What use case you're trying to address and why this PR addresses it. I assume this is for FRC, but is it for testing or to serve some other requirement?
>    2. What testing have you done to verify that these changes are buildable and do not break the academic mainline?
> 3. link to any NI-internal work items that track this work.

Hi @amstewart ,
I have Squashed my commits in the PR and also added commit and PR description as mentioned.

amstewart commented:
We've put in commits to the academic mainline which update the kernel hash. Could you rebase this PR upon `nilrt-academic/master/sumo` and resolve the conflicts?

Commit:
Added support for qemu-zynq7

The current codebase supports only Xilinx-zynq device, current changes will enable support for armv7 based qemu-zynq7 VM.

Signed-off-by: Raagul S <raagul.s@ni.com>

ghost commented:
> We've put in commits to the academic mainline which update the kernel hash. Could you rebase this PR upon `nilrt-academic/master/sumo` and resolve the conflicts?
Hi @amstewart I have rebased the PR and resolved the conflicts

amstewart commented:
Thanks. I've created a fork of upstream [`meta-xilinx as github.com/ni/meta-xilinx:nilrt-academic/master/sumo`](https://github.com/ni/meta-xilinx/tree/nilrt-academic/master/sumo). Once Hershberger's V&V is complete, I'll merge this PR and update the project submodules and components to accommodate the new layer.

jhershbe commented:
When I was build testing this, I needed to add the following to the local.conf.sample

`DEFAULTTUNE_qemu-zynq7 = "cortexa9-vfpv3"`

Without it, it seemed to be building a hard-float version of everything for qemu.

amstewart commented:
@Raagul1 Instead of doing a merge, could you do a rebase on the `nilrt/master/hardknott` commit. I expect that all of my ffmpeg commits, and the merge-commits should drop away, and this PR should just be your authored commits.

ghost commented:
> @Raagul1 Instead of doing a merge, could you do a rebase on the `nilrt/master/hardknott` commit. I expect that all of my ffmpeg commits, and the merge-commits should drop away, and this PR should just be your authored commits.

Hi @amstewart I rebased the commit

ghost commented:
> When I was build testing this, I needed to add the following to the local.conf.sample
> 
> `DEFAULTTUNE_qemu-zynq7 = "cortexa9-vfpv3"`
> 
> Without it, it seemed to be building a hard-float version of everything for qemu.

Hi @jhershbe I included DEFAULTTUNE for qemu-zynq7 in local.conf.sample file

amstewart commented:
@jhershbe Has validated that this PR doesn't adversely affect functionality of the academic mainline. I'll add the meta-xilinx layer to the top-level project.

amstewart commented:
https://github.com/ni/nilrt/pull/196
